doc/source/whatsnew/v1.5.0.rst

```diff
@@ -634,7 +634,8 @@ Categorical
 ^^^^^^^^^^^
 - Bug in :meth:`Categorical.view` not accepting integer dtypes (:issue:`25464`)
 - Bug in :meth:`CategoricalIndex.union` when the index's categories are integer-dtype and the index contains ``NaN`` values incorrectly raising instead of casting to ``float64`` (:issue:`45362`)
--
+- Bug in :meth:`.DataFrameGroupBy.value_counts` includes non-observed categories of non-grouping columns regardless of ``observed`` (:issue:`46357`)
```

Comment:
ok since this is changing non-trivially, can you make a sub-section (ok here is fine, where you show the prior and the fixed behavior)

Reply:
I added a sub-section in notable fixes

Reply:
Sub-section looks good; need to remove this line.